packages/grid/x-data-grid/src/components/GridRow.tsx

```diff
@@ -454,21 +502,30 @@ const GridRow = React.forwardRef<HTMLDivElement, GridRowProps>(function GridRow(
       }
     : null;
 
+  const scrollbarWidth = dimensions.hasScrollY ? dimensions.scrollbarSize : 0;
+  const expandedWidth =
+    dimensions.viewportOuterSize.width - dimensions.columnsTotalWidth - scrollbarWidth;
+  const emptyCellWidth = Math.max(scrollbarWidth, expandedWidth);
```

Comment:
This causes an issue with pinned columns: https://deploy-preview-10059--material-ui-x.netlify.app/x/react-data-grid/column-pinning/

<div class='graphite__hidden'>
          <div>🎥 Video uploaded on Graphite:</div>
            <a href="https://app.graphite.dev/media/video/HUAcyHFPUPbbXFJX5Kwj/9b991c14-e316-4ed6-a15e-ce99c4d56fdc.mov">
              <img src="https://app.graphite.dev/api/v1/graphite/video/thumbnail/HUAcyHFPUPbbXFJX5Kwj/9b991c14-e316-4ed6-a15e-ce99c4d56fdc.mov">
            </a>
          </div>
<video src="https://graphite-user-uploaded-assets-prod.s3.amazonaws.com/HUAcyHFPUPbbXFJX5Kwj/9b991c14-e316-4ed6-a15e-ce99c4d56fdc.mov">Screen Recording 2023-12-20 at 10.53.19.mov</video>


Why is `scrollbarWidth` used here? Shouldn't it be `Math.max(0, expandedWidth);`?

Reply:
Fixed. Not sure why it was written that way (past me had intentions forgotten to me), but the filler cell needs to act differently depending if there's pinned-right columns or not.